src/webnn_native/dml/GraphDML.cpp

```diff
@@ -516,7 +520,9 @@ namespace webnn_native { namespace dml {
                                            constant->GetByteLength());
         mExpression.insert(std::make_pair(constant->PrimaryOutput(), dmlConstant));
         mConstantSet.insert(constant->PrimaryOutput());
-        DAWN_ASSERT(CheckShape(dmlConstant, constant));
+        if (!CheckShape(dmlConstant, constant)) {
```

Comment:
Make CheckShape to return `MaybeError`, so you can simplify the code like `return CheckShape(dmlConstant, constant);`

Reply:
OK, thanks, will update as suggested! @fujunwei